pseudospectral/spectra/derivative_1D.py

```diff
@@ -96,4 +96,9 @@ def scalar_product(self, lhs, rhs, input_basis="real"):
         # (that's why rhs @ lhs and not lhs @ rhs).
         # Furthermore, the @ operator interpretes multi-dimensional objects as "stacks of matrices"
         # and accordingly acts on the LAST index of the first but the SECOND TO LAST index of the second.
-        return rhs @ lhs.transpose(-1, -2).conjugate() * self.a
+        if input_basis == "real":
+            return rhs @ lhs.transpose(-1, -2).conjugate() * self.a
+        elif input_basis == "spectral":
+            return rhs @ lhs.transpose().conjugate()
```

Comment:
I'd say either
```suggestion
            return rhs @ lhs.transpose().conjugate() * self.a
        elif input_basis == "spectral":
            return rhs @ lhs.transpose().conjugate()
```
or
```suggestion
            return rhs @ lhs.transpose(-1, -2).conjugate() * self.a
        elif input_basis == "spectral":
            return rhs @ lhs.transpose(-1, -2).conjugate()
```
it's up to you.